DataFormats/Common/interface/DetSetVectorNew.h

```diff
@@ -212,13 +275,106 @@ namespace edmNew {
       typename DetSetVector<T>::Item & item;
       bool saveEmpty;
     };
+
+    /* fill on demand a given  DetSet
+     */
+    class TSFastFiller {
+    public:
+      typedef typename DetSetVector<T>::data_type value_type;
+      typedef typename DetSetVector<T>::id_type key_type;
+      typedef typename DetSetVector<T>::id_type id_type;
+      typedef typename DetSetVector<T>::size_type size_type;
+
+#ifdef USE_ATOMIC
+      static DetSetVector<T>::Item & dummy() {
+        static  DetSetVector<T>::Item d; return d;
+      }
+      TSFastFiller(DetSetVector<T> & iv, id_type id) :
+        v(iv), item(v.ready()? v.push_back(id): dummy()) { assert(v.filling==true); v.filling = false;}
```

Comment:
if v.filling == true   this TSFastFiller constructor  will fail throwing an exception...

I think there is hidden (undocumented) knowledge:
updateImpl uses the other constructor (just below) as onDemand all Items are supposed to be ready at construction time.
I have to check if this first constructor is really needed: I suspect is there to support Non OnDemand use cases